File: VoiceType/App/AppDependencies.swift
import Foundation
import Observation
import SwiftData

/// 组装全部服务，作为 environment 注入视图树。
@MainActor
@Observable
final class AppDependencies {
    /// 单例：App 场景重建会重复执行 @State 初始值表达式，
    /// 必须保证容器与快捷键只初始化一次
    static let shared = AppDependencies()

    let state: AppState
    let container: ModelContainer
    let history: HistoryStore
    let dictation: DictationController
    let asr: AsrService
    let polish: PolishService

    private init() {
        state = AppState()
        container = try! ModelContainer(for: TranscriptRecord.self)
        history = HistoryStore(container: container)
        asr = AsrService()
        polish = PolishService()
        dictation = DictationController(state: state, asr: asr, history: history, polish: polish)

        HotkeyManager.shared.onHotkey = { [dictation] in dictation.toggle() }
        HotkeyManager.shared.register(SettingsStore.keyCombo)
        asr.warmUp()
        if SettingsStore.polishEnabled {
            polish.warmUp()
        }
    }
}

File: windows/src/VoiceType/App/AppDependencies.cs
using VoiceType.Services;
using VoiceType.UI;

namespace VoiceType.Core;

/// <summary>
/// 组装全部服务（对应 macOS AppDependencies），进程内单例。
/// 在 UI 线程构造（AudioRecorder 捕获同步上下文、HudController 需要 Dispatcher）。
/// </summary>
public sealed class AppDependencies
{
    public static AppDependencies Shared { get; } = new();

    public AppState State { get; }
    public SettingsStore Settings { get; }
    public HistoryStore History { get; }
    public AsrService Asr { get; }
    public PolishService Polish { get; }
    public DictationController Dictation { get; }
    public MeetingController Meeting { get; }

    private AppDependencies()
    {
        Settings = SettingsStore.Default;
        State = new AppState();
        History = new HistoryStore();
        Asr = new AsrService(Settings);
        Polish = new PolishService();
        Dictation = new DictationController(State, Asr, History, Polish, Settings);
        Meeting = new MeetingController(State, Asr);

        HotkeyManager.Shared.OnHotkey += Dictation.Toggle;
        _ = HotkeyManager.Shared.Register(Settings.KeyCombo);
        Asr.WarmUp();
        if (Settings.PolishEnabled)
            Polish.WarmUp();
    }
}
